dotnet/src/SemanticKernel.Abstractions/AI/ChatCompletion/AuthorRole.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.ComponentModel;
+using Microsoft.SemanticKernel.Diagnostics;
+
+namespace Microsoft.SemanticKernel.AI.ChatCompletion;
+
+/// <summary>
+/// A description of the intended purpose of a message within a chat completions interaction.
+/// </summary>
+public readonly struct AuthorRole : IEquatable<AuthorRole>
+{
+    /// <summary>
+    /// The role that instructs or sets the behavior of the assistant.
+    /// </summary>
+    public static readonly AuthorRole System = new("system");
+    /// <summary>
+    /// The role that provides responses to system-instructed, user-prompted input.
+    /// </summary>
+    public static readonly AuthorRole Assistant = new("assistant");
+    /// <summary>
+    /// The role that provides input for chat completions.
+    /// </summary>
+    public static readonly AuthorRole User = new("user");
+
+    /// <summary>
+    /// Gets the label associated with this AuthorRole.
+    /// </summary>
+    /// <remarks>
+    /// The label is what will be serialized into the "role" message field of the Chat Message format.
+    /// </remarks>
+    public string Label { get; }
+
+    /// <summary>
+    /// Creates a new AuthorRole instance with the provided label.
+    /// </summary>
+    /// <param name="label"></param>
+    public AuthorRole(string label)
+    {
+        Verify.NotNull(label, nameof(label));
+        this.Label = label!;
+    }
+
+    /// <summary>
+    /// Returns a value indicating whether two AuthorRole instances are equivalent, as determined by a
+    /// case-insensitive comparison of their labels.
+    /// </summary>
+    /// <param name="left"> the first AuthorRole instance to compare </param>
+    /// <param name="right"> the second AuthorRole instance to compare </param>
+    /// <returns> true if left and right are both null or have equivalent labels; false otherwise </returns>
+    public static bool operator ==(AuthorRole left, AuthorRole right)
+    {
+        if (Object.ReferenceEquals(left, null) != Object.ReferenceEquals(right, null))
+        {
+            return false;
+        }
+        else if (Object.ReferenceEquals(left, null))
+        {
+            return true;
+        }
```

Comment:
I know this is boilerplate, and what I'm about to say matters not one bit... but in the interest of logic puzzles. Is this the same as?:
```suggestion
        if (Object.ReferenceEquals(left, null))
        {
            return Object.ReferenceEquals(right, null);
        }
        
        return left.Equals(right);
```

Reply:
(note: the above diff suggestion is missing the final return in the red/removed part)...

 `else
        {
            return left.Equals(right);
        }
    }
    `